TF-14957 Add Team Tokens Index API (#667)
hashicorp/terraform-docs-common: website/docs/cloud-docs/api-docs/organization-team-tokens.mdx

---

[200]: https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/200

[201]: https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/201

[202]: https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/202

[204]: https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/204

[400]: https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/400

[401]: https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/401

[403]: https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/403

[404]: https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/404

[409]: https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/409

[412]: https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/412

[422]: https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/422

[429]: https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/429

[500]: https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/500

[504]: https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/504

[JSON API document]: /terraform/cloud-docs/api-docs#json-api-documents

[JSON API error object]: https://jsonapi.org/format/#error-objects

# Team Tokens API

## List Team Tokens

> List team tokens in an organization.

Reviewer: ```suggestion
Lists the [team tokens](/terraform/cloud-docs/users-teams-organizations/teams#api-tokens) in an organization.
```

`GET organizations/:organization_name/team-tokens`

| Parameter            | Description                                              |
|----------------------|----------------------------------------------------------|

> | `:organization_name` | The name of the organization the team tokens belong to |

Reviewer: ```suggestion
| `:organization_name` | The name of the organization whose team tokens you want to list. |
```

> The objects returned by this endpoint only contain metadata, and do not include the secret text of any authentication tokens. A token is only shown upon creation, and cannot be recovered later.

Reviewer: ```suggestion
This endpoint returns object metadata and does not include secret authentication details of any token. You can only view a token when you create it and cannot recover it later.
```

> By default, the tokens returned are ordered by ascending expiration date.

Reviewer: ```suggestion
By default, this endpoint returns tokens by ascending expiration date.
```

| Status  | Response                                                | Reason                                                                                |
| ------- | ------------------------------------------------------- | ------------------------------------------------------------------------------------- |
| [200][] | [JSON API document][] (`type: "team-tokens"`)           | The request was successful                                                            |

> | [200][] | Empty [JSON API document][]                             | Organization has no team authentication tokens                                        |

Reviewer: ```suggestion
| [200][] | Empty [JSON API document][]                             | The specified organization has no team tokens                                        |
```
